src/lib/ares_options.c

```diff
@@ -474,6 +482,19 @@ ares_status_t ares__init_by_options(ares_channel_t            *channel,
     }
   }
 
+  /* Set fields for server failover behavior */
+  if (optmask & ARES_OPT_SERVER_FAILOVER) {
+    if (options->server_failover_opts.retry_chance == 0 ||
```

Comment:
maybe we should allow `retry_chance` of 0 to disable this logic?

Reply:
I think that is sensible, and the code in `ares__failover_server` is already resilient to the retry chance being 0. I have updated the options validation to allow this (and updated the man page docs to describe it). Thanks.